src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -635,7 +636,7 @@ BinderCheckTypeResult ReportInvalidOperation()
 
                         default:
                             // Time + number = Time
-                            var resRight = CheckTypeCore(errorContainer, node.Right, rightType, DType.Number, /* coerced: */ DType.String, DType.Boolean);
+                            var resRight = CheckTypeCore(errorContainer, node.Right, rightType, DType.Number, /* coerced: */ DType.String, DType.Boolean, DType.UntypedObject);
```

Comment:
>, DType.UntypedObject [](http://example.com/codeflow?start=152&length=21)

We should have tests for those scenarios:

- Time + untyped (and vice-versa)
- Date + untyped (and vice-versa)
- DateTime + untyped (and vice-versa)